rpc/Base.ts

```diff
@@ -91,24 +95,39 @@ export abstract class Client<
   >(
     methodName: MethodName,
     params: Parameters<Config_["RpcMethods"][MethodName]>,
-  ): Promise<IngressMessage> => {
+  ): Promise<IngressMessage | InternalError> => {
     const init = <msg.InitMessage<Config_, MethodName>> {
       jsonrpc: "2.0",
       id: this.uid(),
       method: methodName,
       params,
     };
     const isCorrespondingRes = IsCorrespondingRes(init);
-    const ingressMessagePending = deferred<IngressMessage>();
-    this.listen((stopListening) => {
-      return (res) => {
-        if (isCorrespondingRes(res)) {
+    const ingressMessagePending = deferred<IngressMessage | InternalError>();
+    let stopListening: () => void = () => {};
+    this.listen(
+      (stop) => {
+        stopListening = stop;
+        return (res) => {
+          if (isCorrespondingRes(res)) {
+            stop();
+            ingressMessagePending.resolve(res as IngressMessage);
+          }
+        };
+      },
+      (stopListening) =>
+        (error) => {
+          stopListening();
+          ingressMessagePending.resolve(error);
+        },
+    );
+    this.send(init)
+      .then((sendResult) => {
+        if (sendResult instanceof Error) {
           stopListening();
-          ingressMessagePending.resolve(res as IngressMessage);
+          ingressMessagePending.resolve(sendResult);
         }
-      };
-    });
-    this.send(init);
+      });
```

Comment:
the `async/await` was causing an error in the extrinsic integration tests.
For those tests, there seem to be some synchronization issue where the first subscription message is lost.